RadixWallet/Core/SharedModels/Entities/EntitiesControlledByFactorSource.swift

```diff
@@ -24,6 +24,55 @@ public struct EntitiesControlledByFactorSource: Sendable, Hashable, Identifiable
 }
 
 extension EntitiesControlledByFactorSource {
+	public struct PerCurve: Equatable, Sendable {
+		public struct ID: Sendable, Hashable {
+			public let factorSourceID: FactorSource.ID.FromHash
+			public let isOlympia: Bool
+		}
+
+		public let id: ID
+		public let accounts: NonEmpty<IdentifiedArrayOf<Profile.Network.Account>>
+		public let hiddenAccounts: NonEmpty<IdentifiedArrayOf<Profile.Network.Account>>?
+	}
+
+	public var olympia: PerCurve? {
+		guard let olympiaAccounts else { return nil }
```

Comment:
we return `nil` if the accounts for this curve is empty, because if a user has "added Olympia Seed Phrase" where the mnemonic was already a BDFS -> the `cryptoParameters` now contain both sets of Curve. But we dont wanna show the Seed Phrase twice in `App Settings -> Account Sec -> Seed Phrases` if it is not yet used by any  Olympia Accounts....

Reply:
But is it fine that it's `nil` even if there **are** hidden accounts? This just checks for non-hidden accounts.

Reply:
In general there is a bit of overlap between all these computed vars. 

`self.babylon.accounts` is the same as `self.babylonAccounts`

`self.babylon.hiddenAccounts` is **almost** the same as `self.babylonAccountsHidden`

Reply:
we dont care about the hidden accounts, which is why the check is for non-hidden accounts.

Yes some overlap, not the greatest, not the worst. We have bigger fish to fry and I hope we can refactor this code soon anyway.